dev/com.ibm.ws.ejbcontainer.mdb.ra_fat/test-applications/MsgEndpointEJB.jar/src/com/ibm/ws/ejbcontainer/fat/msgendpoint/ejb/MDBTimedCMTBean.java

```diff
@@ -404,6 +396,11 @@ public void onMessage(Message msg) {
                         // 23 - Verify NoSuchObjectLocalException occurs accessing self cancelled
                         //      timer
                         // -----------------------------------------------------------------------
+                        svTimerIntervalLatch = new CountDownLatch(2);
```

Comment:
I assume the latch is set to 2 because there are 2 timers remaining.  However, there appears to be a very narrow window where a timer thread could be performing a countdown at exactly the same time the new latch is created, but before the next line where cancel is true... This could lead to the countdown latch tripping, and yet a timer still exists.  Making both svTimerIntervalLatch and svCancelTimer volatile might help reduce the window further, but won't completely eliminate that.  Reversing the order doesn't really help either.

Adding a sync block around these 2 lines (i.e. creating the latch and setting cancel to true), and then sync'ing on the same object when performing the countdown in the timeout method could eliminte the window; but it seems pretty unlikely it would ever be hit.